src/main/java/net/minecraftforge/fml/AutomaticEventSubscriber.java

```diff
@@ -54,16 +55,21 @@ public static void inject(final ModContainer mod, final ModFileScanData scanData
                 collect(Collectors.toList());
 
         ebsTargets.forEach(ad -> {
-            @SuppressWarnings("unchecked")
-            final List<ModAnnotation.EnumHolder> sidesValue = (List<ModAnnotation.EnumHolder>)ad.getAnnotationData().
-                    getOrDefault("value", Arrays.asList(new ModAnnotation.EnumHolder(null, "CLIENT"), new ModAnnotation.EnumHolder(null, "DEDICATED_SERVER")));
-            final EnumSet<Dist> sides = sidesValue.stream().map(eh -> Dist.valueOf(eh.getValue())).
-                    collect(Collectors.toCollection(() -> EnumSet.noneOf(Dist.class)));
+            final EnumSet<Dist> sides = get(ad, "values", Dist.class, Dist.values());
+            final EnumSet<Mod.EventBusSubscriber.Target> buses = get(ad, "bus", Mod.EventBusSubscriber.Target.class, Mod.EventBusSubscriber.Target.values());
             final String modId = (String)ad.getAnnotationData().getOrDefault("modId", mod.getModId());
```

Comment:
This line needs fixing as well. The map key should be all lowercase. If it is missing, then we need to make sure the annotated class has the same type as `mod.getMod()`.

Reply:
nice catch